### DISCUSSION TITLE
Seq Okta Integration (email_verified)

### DISCUSSION THREAD

**Discussion by specialagentsoftware** [datalust/seq-tickets#1620] [Help] [answered]
After correcting some network issues, after attempting to do an Okta authentication I got the following message. 

[22:23:24 WRN] Identity successfully authenticated, but the OpenID Connect provider did not supply the `email_verified` claim.
[22:23:24 WRN] Identity "00uvb7ttdoExfjOiD0x7" successfully authenticated, but the email claim associated with it is not verified.

Is this setting within okta? or is there a way to set this within seq?

**Comment by nblumhardt** [comment, marked as answer]
Hi @specialagentsoftware, thanks for getting in touch. Seq currently requires the `email_verified` claim when matching accounts by email address; it's a standard field in the `email` OIDC scope, but it appears to be unsupported by Okta.

Is it possible to confirm with Okta support that there's no way to have that claim included? Having Okta supply the claim would be the fastest way to get unblocked.

If that turns out to be impossible, we'd be open to adding an "ignore `email_verified` claim" option in our next maintenance release, but the timeline for that is not set, currently (our last maintenance release was just a few days ago).

Hope this helps,
Nick

**Reply by specialagentsoftware** [reply]
We reached out to Okta and they suggested some changes on their side. The changes they had us make got the email_verified claim included the id_token.. but it does now show up in the token. So Seq still has an issue with the verification of the email. It does not appear that you can use domain\user or any other info with OPENID other than a verified email. Do you have any other suggestions of anything else we can try?

**Reply by nblumhardt** [reply]
Hi @specialagentsoftware; thanks for the update. Would it be possible to collect the error/warning from Seq's log again, now that the claim is being supplied? This should point us towards the best thing to try next.

**Reply by specialagentsoftware** [reply]
@nblumhardt 
{"@t":"2022-07-21T11:58:51.6290882Z","@mt":"Identity successfully authenticated, but the OpenID Connect provider did not supply the `email_verified` claim.","@l":"Warning","Principal":{"Id":"principal-public","OnBehalfOfUserId":null},"ActionId":"f5d4b780-2783-47e6-b00b-90301639fba5","ActionName":"Seq.Server.Web.UI.OpenIdConnectController.ChallengeGet (Seq)","RequestId":"40000d4f-0002-fa00-b63f-84710c7967bb","RequestPath":"/oidc/challenge"}

{"@t":"2022-07-21T11:58:51.6292238Z","@mt":"Identity {UniqueIdentifier} successfully authenticated, but the email claim associated with it is not verified.","@l":"Warning","UniqueIdentifier":"00uvb7ttdoExfjOiD0x7","Principal":{"Id":"principal-public","OnBehalfOfUserId":null},"ActionId":"f5d4b780-2783-47e6-b00b-90301639fba5","ActionName":"Seq.Server.Web.UI.OpenIdConnectController.ChallengeGet (Seq)","RequestId":"40000d4f-0002-fa00-b63f-84710c7967bb","RequestPath":"/oidc/challenge"}

**Reply by specialagentsoftware** [reply]
@nblumhardt 
As I said before, email_verified appears in id_token.. but it is not present in token.

**Reply by specialagentsoftware** [reply]
@nblumhardt 
Any update on this? I also have included the id_token payload that is coming back from okta after our change. Does that claim need to be present in token as well? Because I do not see it. 
![1658427872123](https://user-images.githubusercontent.com/8382528/181005937-894e3512-abc4-4280-be56-93914d591625.png)
?

**Reply by specialagentsoftware** [reply]
@nblumhardt 
We purchased an enterprise license. So if there is some other communication method I should be using, I am happy to do that as well.

**Reply by nblumhardt** [reply]
Hi Jeremy,

Sorry about the delay in getting back to you; `support@datalust.co` can sometimes be a better channel for priority support, but in this case it was my mistake, I'd meant to send an update yesterday and dropped the ball.

To the best of my knowledge the claim on `id_token` should work, but there's a good chance I'm wrong. In any case we'd prefer to have some option out-of-the-box to get this going.

We have an opportunity to produce a patch build in the next few days; in it we'll include a feature flag that will skip email address validation checking when enabled (some other applications appear to take this approach).

In the meantime, you may also have some success with a related flag:

If you're using Docker, set the following environment variable and restart the container:

```
SEQ_FEATURES_ENABLED=OidcSaveTokens
```

Or on Windows:

```
seq config set -k features.enabled -v OidcSaveTokens
seq service restart
```

Let me know if the above is successful.

It would be great if you can drop us a line via email, so that I can reply privately with a link to the patch as soon as it is built.

Best regards,
Nick

**Reply by nblumhardt** [reply]
Sorry @specialagentsoftware - just one more thought, if you haven't tried logging in via a fresh browser session (different browser or private window) this might also sort out the ignored claim.

**Reply by specialagentsoftware** [reply]
@nblumhardt Thanks for the info. I tried the above config change and then started a new browser session and I encountered the same error. I will send an email to the address you shared so that you can send me a patch build link there. Thank you.

**Reply by specialagentsoftware** [reply]
@nblumhardt Email sent.

**Reply by nblumhardt** [reply]
Received! Thank you 👍

**Reply by specialagentsoftware** [reply]
@nblumhardt I had seen this issue as well. https://github.com/datalust/seq-tickets/issues/1154.. it was related to not being able to get email in the initial token for IdentityServer4... would enabling that allow extra info to be collected from Okta? I am totally fine with getting a build that allows us to disable the email_verified claim. Just thought I would mention it. We are currently running 2022.1.7929. It appears this was already done and looks like it would have been included already in a build. I will leave this here just in case I have misunderstood.

**Reply by nblumhardt** [reply]
Thanks for the note; yes, this is already enabled in 2022.1.7929 :+1:

**Reply by juarola** [reply]
Hello! A bit of an old topic, but i'm seeing these entries in logs when attempting to login via Ubisecure SSO using OIDC:

```
{"@t":"2025-10-01T08:36:11.3560032Z","@sp":"c8f5f887c3b647e4","@tr":"8ef2a26b4726b3b5d6987c60cf508e75","@mt":"Identity successfully authenticated, but the OpenID Connect provider did not supply the `email_verified` claim.","@l":"Warning","Principal":{"Id":"principal-public","OnBehalfOfUserId":null},"ActionId":"87a7d261-f2f6-4bc4-89da-76111b5f3a11","ActionName":"Seq.Server.Web.UI.OpenIdConnectController.ChallengeGet (Seq)","RequestId":"0HNG0OFHQF0CR:00000035","RequestPath":"/oidc/challenge","ConnectionId":"0HNG0OFHQF0CR"}
{"@t":"2025-10-01T08:36:11.3561881Z","@sp":"c8f5f887c3b647e4","@tr":"8ef2a26b4726b3b5d6987c60cf508e75","@mt":"Identity {UniqueIdentifier} successfully authenticated, but the email claim associated with it is not verified.","@l":"Warning","UniqueIdentifier":"cn=96e346d6-c312-4e42-a60f-8a3b193eb61f,ou=Users,ou=eIDM Users,cn=Ubilogin,dc=amdev,dc=apptoyou,dc=com","Principal":{"Id":"principal-public","OnBehalfOfUserId":null},"ActionId":"87a7d261-f2f6-4bc4-89da-76111b5f3a11","ActionName":"Seq.Server.Web.UI.OpenIdConnectController.ChallengeGet (Seq)","RequestId":"0HNG0OFHQF0CR:00000035","RequestPath":"/oidc/challenge","ConnectionId":"0HNG0OFHQF0CR"}
```

I think i sort of dropped off the wagon, so was there some fix for this? It seems that i'm not able to get "email_verified"-claim with value of "true" sent from the SSO, so is there some workaround? I cannot find "GetClaimsFromUserInfoEndpoint" mentioned in 1154 from docs @ https://datalust.co/docs/json-config

---------
EDIT: It was just my Ubisecure config-fu that was failing, everything working fine now after i got the email_verified-claim setup correctly!
